src/modules/powerrename/lib/Settings.cpp

```diff
@@ -1,171 +1,60 @@
-}
-
-bool CSettings::SetEnabled(_In_ bool enabled)
-{
-    return SetRegBoolValue(c_enabled, enabled);
-}
-
-bool CSettings::GetShowIconOnMenu()
-{
-    return GetRegBoolValue(c_showIconOnMenu, c_showIconOnMenuDefault);
-}
-
-bool CSettings::SetShowIconOnMenu(_In_ bool show)
-{
-    return SetRegBoolValue(c_showIconOnMenu, show);
-}
-
-bool CSettings::GetExtendedContextMenuOnly()
-{
-    return GetRegBoolValue(c_extendedContextMenuOnly, c_extendedContextMenuOnlyDefaut);
-}
-
-bool CSettings::SetExtendedContextMenuOnly(_In_ bool extendedOnly)
-{
-    return SetRegBoolValue(c_extendedContextMenuOnly, extendedOnly);
-}
-
-bool CSettings::GetPersistState()
-{
-    return GetRegBoolValue(c_persistState, c_persistStateDefault);
-}
-
-bool CSettings::SetPersistState(_In_ bool persistState)
-{
-    return SetRegBoolValue(c_persistState, persistState);
-}
-
-bool CSettings::GetMRUEnabled()
-{
-    return GetRegBoolValue(c_mruEnabled, c_mruEnabledDefault);
-}
-
-bool CSettings::SetMRUEnabled(_In_ bool enabled)
-{
-    return SetRegBoolValue(c_mruEnabled, enabled);
-}
-
-DWORD CSettings::GetMaxMRUSize()
-{
-    return GetRegDWORDValue(c_maxMRUSize, c_maxMRUSizeDefault);
-}
-
-bool CSettings::SetMaxMRUSize(_In_ DWORD maxMRUSize)
-{
-    return SetRegDWORDValue(c_maxMRUSize, maxMRUSize);
-}
-
-DWORD CSettings::GetFlags()
-{
-    return GetRegDWORDValue(c_flags, c_flagsDefault);
-}
-
-bool CSettings::SetFlags(_In_ DWORD flags)
-{
-    return SetRegDWORDValue(c_flags, flags);
-}
-
-bool CSettings::GetSearchText(__out_ecount(cchBuf) PWSTR text, DWORD cchBuf)
-{
-    return GetRegStringValue(c_searchText, text, cchBuf);
-}
-
-bool CSettings::SetSearchText(_In_ PCWSTR text)
-{
-    return SetRegStringValue(c_searchText, text);
-}
-
-bool CSettings::GetReplaceText(__out_ecount(cchBuf) PWSTR text, DWORD cchBuf)
-{
-    return GetRegStringValue(c_replaceText, text, cchBuf);
-}
-
-bool CSettings::SetReplaceText(_In_ PCWSTR text)
-{
-    return SetRegStringValue(c_replaceText, text);
-}
+#include <filesystem>
+#include <commctrl.h>
 
-bool CSettings::SetRegBoolValue(_In_ PCWSTR valueName, _In_ bool value)
+namespace
 {
-    DWORD dwValue = value ? 1 : 0;
-    return SetRegDWORDValue(valueName, dwValue);
-}
+    const wchar_t c_powerRenameDataFilePath[] = L"power-rename-settings.json";
 
-bool CSettings::GetRegBoolValue(_In_ PCWSTR valueName, _In_ bool defaultValue)
-{
-    DWORD value = GetRegDWORDValue(valueName, (defaultValue == 0) ? false : true);
-    return (value == 0) ? false : true;
-}
+    const wchar_t c_rootRegPath[] = L"Software\\Microsoft\\PowerRename";
+    const wchar_t c_mruSearchRegPath[] = L"SearchMRU";
+    const wchar_t c_mruReplaceRegPath[] = L"ReplaceMRU";
 
-bool CSettings::SetRegDWORDValue(_In_ PCWSTR valueName, _In_ DWORD value)
-{
-    return (SUCCEEDED(HRESULT_FROM_WIN32(SHSetValue(HKEY_CURRENT_USER, c_rootRegPath, valueName, REG_DWORD, &value, sizeof(value)))));
-}
+    const wchar_t c_enabled[] = L"Enabled";

```

Comment:
Should we be consistent with FZ settings and use dash-case instead of UpperCamelCase?

Reply:
It complicates migration a bit, since we need to handle different format, but it can be done.

Reply:
Please keep the current UpperCamelCase, dash-case is just a bad idea to begin with and we will very likely remove it from FZ.